workflow/controller/controller.go

```diff
@@ -789,6 +795,32 @@ func (wfc *WorkflowController) runWorker() {
 	}
 }
 
+func (wfc *WorkflowController) runArchiveWorker() {
+	ctx := context.TODO()
+	for wfc.processNextArchiveItem(ctx) {
+	}
+}
+
+func (wfc *WorkflowController) processNextArchiveItem(ctx context.Context) bool {
```

Comment:
This should be below `processNextItem`, matching the ordering of `runWorker` and `runArchiveWorker`